examples/nextjs-spa/src/pages/recovery.tsx

```diff
@@ -15,69 +15,51 @@ import { AxiosError } from "axios"
 // We will use UserAuthCard from Ory Elements to display the recovery form.
 import Layout from "@/components/layout"
 import { UserAuthCard } from "@ory/elements"
-import { HandleError } from "../pkg/hooks"
+import { HandleError } from "@/pkg/hooks"
+import { SetUriFlow } from "@/pkg/helpers"
 import { NextPageWithLayout } from "./_app"
 
 const Recovery: NextPageWithLayout = () => {
   const [flow, setFlow] = useState<RecoveryFlow>()
 
   // Get flow information from the URL
   const router = useRouter()
-  const handleError = HandleError()
 
   const flowId = String(router.query.flow || "")
   const returnTo = String(router.query.return_to || "")
 
-  const getRecoveryFlow = useCallback(
+  const getFlow = useCallback(
     (id: string) =>
       ory
         .getRecoveryFlow({ id })
         .then(({ data }) => {
           setFlow(data)
         })
-        .catch((err: AxiosError) => handleError(err))
-        .catch((err: AxiosError) => {
-          switch (err.response?.status) {
-            case 410:
-              // Status code 410 means the request has expired - so let's load a fresh flow!
-              router.push("/recovery")
-            case 403:
-              // Status code 403 implies some other issue (e.g. CSRF) - let's reload!
-              router.push("/recovery")
-            default:
-              router.push({
-                pathname: "/error",
-                query: {
-                  error: JSON.stringify(err, null, 2),
-                  id: err.response?.data.error?.id,
-                  flowType: router.pathname,
-                },
-              })
-          }
-        }),
-    [handleError, router],
+        .catch(handleError),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm not sure if we should disable linter rules like this.
Since the functions are memoized through the `useCallback` function, it should be alright to add them to this dependency list.